### PR TITLE
Added database in fqdn for dbt

### LINKED ISSUE #4460: DBT - create fqn alongwith database name
DBT right now matches with schema and table names.
Let's add a support for database name in the fqn.

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix #4460
#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 1]
```diff
@@ -603,7 +606,7 @@ def _parse_data_model(self):
                         columns=columns,
                         upstream=upstream_nodes,
                     )
-                    model_fqdn = f"{schema}.{model_name}".lower()
+                    model_fqdn = f"{database}{FQDN_SEPARATOR}{schema}{FQDN_SEPARATOR}{model_name}".lower()
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/sql_source.py:609]: @ulixius9 why are we not using a method to construct this FQN?

Comment (harshach) [reply ingestion/src/metadata/ingestion/source/sql_source.py:609]: There are several places we are not using FQN in sql_source.py. Lets fix these.
@pmbrull lets see if we can merge this https://github.com/open-metadata/OpenMetadata/pull/4134 and use antlr backend and UI is already on it

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:609]: https://github.com/open-metadata/OpenMetadata/issues/4496

**Review by ayush-shah — APPROVED** [review]

**Comment by ayush-shah** [comment]
Hey @OnkarVO7 can you fix the errors from sonar and then squash + merge this PR?

**Pull request merged**